### PR TITLE
CLOUDP-57847: Implement global alerts resource in ops manager client

### PR DESCRIPTION
**Pull request by andreaangiolillo**
_Jira ticket:_ [CLOUDP-57847](https://jira.mongodb.org/browse/CLOUDP-57847)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] **I have tested it using my local ops manager**

### CODE REVIEW

**Comment by gssbzn** [comment]
@andreaangiolillo why is this waiting for atlas?

File: opsmngr/global_alerts.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+type GlobalAlertsServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+type GlobalAlert struct {
+	atlas.Alert
+	SourceTypeName string       `json:"sourceTypeName,omitempty"`
+	Tags           []string     `json:"tags,omitempty"`
+	Links          []atlas.Link `json:"links,omitempty"`
```
Comment (gssbzn) [inline opsmngr/global_alerts.go]:
```suggestion
	Links          []*atlas.Link `json:"links,omitempty"`
```

File: opsmngr/checkpoints.go [hunk 2]
```diff
@@ -35,7 +35,11 @@ func (s *CheckpointsServiceOp) List(ctx context.Context, groupID, clusterName st
 		return nil, nil, atlas.NewArgError("clusterName", "must be set")
 	}
 
-	path := fmt.Sprintf(checkpoints, groupID, clusterName)
+	basePath := fmt.Sprintf(checkpoints, groupID, clusterName)
+	path, err := setListOptions(basePath, listOptions)
```
Comment (andreaangiolillo) [inline opsmngr/checkpoints.go]: list of options should not be passed as request body but should be inserted in the path

Comment (gssbzn) [reply opsmngr/checkpoints.go]: is this a problem in atlas as well?

File: opsmngr/global_alerts.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+	root := new(GlobalAlert)
+	resp, err := s.Client.Do(ctx, req, root)
+
+	return root, resp, err
+}
+
+// List gets all global alerts.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/global-alerts/
+func (s *GlobalAlertsServiceOp) List(ctx context.Context, opts atlas.AlertsListOptions) (*GlobalAlerts, *atlas.Response, error) {
```
Comment (gssbzn) [inline opsmngr/global_alerts.go]: if this is what's holding you on atlas let's please just make this work and we'll do a separate PR once atlas is updated

**Comment by andreaangiolillo** [comment]
@gssbzn it is using **atlas.AlertsListOptions** which is in https://github.com/mongodb/go-client-mongodb-atlas/pull/69

File: opsmngr/checkpoints.go (hunk 2 above)
Comment (andreaangiolillo) [reply opsmngr/checkpoints.go]: yes, it is. I am going to open a PR on atlas

Comment (gssbzn) [reply opsmngr/checkpoints.go]: add it to this PR https://github.com/mongodb/go-client-mongodb-atlas/pull/69 so we only wait for that one

File: opsmngr/global_alerts.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+type GlobalAlerts struct {
+	Links      []*atlas.Link  `json:"links"`
+	Results    []*GlobalAlert `json:"results"`
+	TotalCount int            `json:"totalCount"`
+}
+
+// AlertsListOptions contains the list of options for Alerts
+type AlertsListOptions struct {
```
Comment (gssbzn) [inline opsmngr/global_alerts.go]: clever solution 👍

File: opsmngr/global_alerts_test.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
+package opsmngr
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+
```
Comment (gssbzn) [inline opsmngr/global_alerts_test.go]: can we remove the extra space?

File: opsmngr/opsmngr.go [hunk 6]
```diff
@@ -195,3 +199,30 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*atl
 	}
 	return response, err
 }
+
+func setListOptions(s string, opt interface{}) (string, error) {
```
Comment (gssbzn) [inline opsmngr/opsmngr.go]: [nit] can we rename this to 
```suggestion
func setQueryParams(s string, opt interface{}) (string, error) {
```
it makes more sense now that we use it for more than lists

File: opsmngr/global_alerts.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+	root := new(GlobalAlert)
+	resp, err := s.Client.Do(ctx, req, root)
+
+	return root, resp, err
+}
+
+// List gets all global alerts.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/global-alerts/
+func (s *GlobalAlertsServiceOp) List(ctx context.Context, opts AlertsListOptions) (*GlobalAlerts, *atlas.Response, error) {
```
Comment (gssbzn) [inline opsmngr/global_alerts.go]: [q] should `opts` be a pointer? what's the signature for alerts?
